pkg/system/phase4_configuring.go

```diff
@@ -682,6 +685,49 @@ func (r *Reconciler) RegisterToCluster() error {
 	return r.NBClient.RegisterToCluster()
 }
 
+// ReconcileDefaultNamespaceStore checks if the noobaa is running on Fusion HCI with
+// spectrum scale and attempts to the default nsfsnamespacestore using spectrum scale
+// storage class.
+func (r *Reconciler) ReconcileDefaultNamespaceStore() error {
+	// Skip if joining another NooBaa
+	if r.JoinSecret != nil {
+		return nil
+	}
+
+	// Check if ODF is installed on Fusion HCI cluster with spectrum scale.
+	if !util.IsFusionHCIWithScale() {
+		r.Logger.Info(" IBM Fusion HCI with spectrum scale NOT detected.")
+		return nil
+	}
+
+	log := r.Logger.WithField("func", "ReconcileDefaultNamespaceStore")
+
+	util.KubeCheck(r.DefaultNamespaceStore)
+	// Namespacestore already exist
+
+	if r.DefaultNamespaceStore.UID != "" {
+		log.Infof("NamespaceStore %s already exists. skipping Reconcile", r.DefaultNamespaceStore.Name)
+		return nil
+	}
+
+	if err := r.prepareDefaultNsfsPvc(); err != nil {
+		return err
+	}
+	r.Own(r.DefaultNsfsPvc)
+	r.Own(r.DefaultNamespaceStore)
+
+	if err := r.Client.Create(r.Ctx, r.DefaultNsfsPvc); err != nil {
+		log.Errorf("got error on DefaultNsfsPvc creation. error: %v", err)
+		// return err
```

Comment:
why is this in comment? is it related to cases we already have the PVC?

Reply:
done